service/src/account_funding.rs

```diff
@@ -92,20 +93,41 @@ pub fn setup_account_funding(
 
 // Alice plays the faucet and sends some funds to the account if balance is low
 fn ensure_account_has_funds(api: &ParentchainApi, accountid: &AccountId32) -> Result<(), Error> {
-	info!("Ensuring funds for Account: {} on chain with genesis hash {}", accountid, api.genesis_hash());
+	info!(
+		"Ensuring funds for Account: {} on chain with genesis hash {}",
+		accountid,
+		api.genesis_hash()
+	);
 	// check account balance
 	let free_balance = api.get_free_balance(accountid)?;
-	info!("TEE's free balance = {:?} (Account: {})", free_balance, accountid);
+	debug!("TEE's free balance = {:?} (Account: {})", free_balance, accountid);
 
 	let existential_deposit = api.get_existential_deposit()?;
-	info!("Existential deposit is = {:?}", existential_deposit);
-
-	let min_required_funds =
-		existential_deposit.saturating_mul(EXISTENTIAL_DEPOSIT_FACTOR_FOR_INIT_FUNDS);
+	debug!("Existential deposit is = {:?}", existential_deposit);
+	let balance_transfer_fee = api
+		.get_fee_details(
+			api.balance_transfer_allow_death(
+				MultiAddress::Address32([0u8; 32]),
+				existential_deposit,
+			)
+			.encode()
+			.into(),
+			None,
+		)
+		.unwrap()
+		.unwrap()
+		.inclusion_fee
+		.unwrap()
+		.inclusion_fee();
+	debug!("Balance Transfer Fee is = {:?}", balance_transfer_fee);
+
+	let min_required_funds = existential_deposit
+		.saturating_mul(EXISTENTIAL_DEPOSIT_FACTOR_FOR_INIT_FUNDS)
+		.saturating_add(balance_transfer_fee.saturating_mul(100));
```

Comment:
this should probably be upstreamed as it is more robust than the previous solution

Reply:
I think that the `.saturating_mul(EXISTENTIAL_DEPOSIT_FACTOR_FOR_INIT_FUNDS)` is not a good way to determine the `min_required_funds`. It has no connection to any onchain metric - it is just an arbitrary value set by us. I think it is better to remove it and increase the factor for the `balance_transfer_fee`, i.e.:

```rust
// ensure that we can send 10_000 transfers before the account falls below the existential deposit.
let min_required_funds = existential_deposit
		.saturating_add(balance_transfer_fee.saturating_mul(10_000));
```

Then we have set a value for `min_required_fund` that we can relate to an actual onchain significance.